marklogic-data-hub-central/ui/src/components/modeling/entity-type-modal/entity-type-modal.tsx

```diff
@@ -32,6 +32,8 @@ const EntityTypeModal: React.FC<Props> = (props) => {
   const [errorMessage, setErrorMessage] = useState("");
   const [loading, toggleLoading] = useState(false);
 
+  const reservedEntityNamesArray = ["Collection", "createdByJob", "createdByStep", "entity-type"];
```

Comment:
Pinging @wooldridge  on this - we really ought to have the backend provide this knowledge to the front end, as the backend understands why these shouldn't be used. I think ideally, this is part of the validation check on the backend when an entity is saved. Worst case, the backend provides this data to the UI for the UI to enforce, but I'd really rather see the backend enforce it, as there could be other interfaces in the future for creating entity types that need this same logic.

Reply:
I can do that instead. If there's a concern that the array were to be consistently updated in the future, that would definitely become tedious and error-prone.

Reply:
@rjrudin, @devinnicholson, definitely important to consider. We already do a client-side validation for bad chars (spaces et al.) and that there's something in the field, and this client-side validation pattern is what UX has prescribed across much of the app. Doing some validation client-side and some server side would be extra work for the frontend and an inconsistent user experience. I see this strategy as manageable for the frontend so I prefer what Devin is doing here.

Reply:
Client-side validation is fine, I'm more interested in where the knowledge of this is stored. Is there an endpoint call that the UI is making here, before the entity is saved, where the backend could include the list of forbidden names? 

My intent here is that Pari can also create entity models via a text editor and then deploying them via Gradle. We need this same validation applied when she does that.

Reply:
@devinnicholson If you can point me to an appropriate endpoint - I'm thinking the one that returns all the entity types - then I can submit a small sub-task PR to enhance that endpoint to include this list of "forbiddenEntityNames". I'll bake that into the backend and get a test in place for it as well. Then the UI can refer to that list and still perform client side validation.

Reply:
@rjrudin let's put this sort of information somewhere in the code base that both the UI and gradle can access directly.

Reply:
There's getPrimaryEntityTypes() found in ModelsService, which I believe is what you're looking for. @rjrudin

Reply:
Oh, to answer your first question, the call when creating an entity is to /api/models passing name and description (createModel) after a client side validation check has passed for naming convention (createEntityType in entity-type-modal.tsx)

Reply:
@rjrudin @devinnicholson I'm fine with adding this info to the primaryEntityTypes endpoint payload.

We'll want to update the mock test data for that endpoint, which I know includes this:

https://github.com/marklogic/marklogic-data-hub/blob/develop/marklogic-data-hub-central/ui/src/assets/mock-data/curation/flows.data.ts#L183

Reply:
I've created https://project.marklogic.com/jira/browse/DHFPROD-7174 and will submit a PR for that this morning.